### PR TITLE
Major Graphics Change #1070

### PR DESCRIPTION
**Pull request by mridulnagpal**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by mridulnagpal** [comment]
@jywarren Please have a look

**Comment by jywarren** [comment]
Hi, can you post a screenshot? Thank you!!!

**Comment by jywarren** [comment]
Oh, i'm sorry, i guess i assumed from the title it'd be visual changes. No need unless there are. Looking through now.

File: app/assets/javascripts/locationForm.js [hunk 1]
```diff
@@ -58,6 +58,13 @@ function addGrid(map)
    _renderCells: function(bounds) {
      var cells = this._cellsInBounds(bounds);
      this.focused_cell_id = cells.length == 0 ? null : cells[0].id;
+     window.gridSquareNWCorner = function(){
```
Comment (jywarren) [inline app/assets/javascripts/locationForm.js]: Hmm, does this have to be attached to the window? That basically makes it a global. Where is it called from?

Comment (mridulnagpal) [inline app/assets/javascripts/locationForm.js]: I don't know the use of this function yet. It was just written to create one function. That's why I made it global.

**Comment by mridulnagpal** [comment]
@jywarren This patch contains the following:
1. A function gridSquareNWCorner() to get north-west corner of the selected cell in the map
2. A form to submit Lat and Long for the center of map
3. Trigger location detection on map pan/zoom displayed below the map
4. Option to obscure your current location and present it on map

**Comment by jywarren** [comment]
Just to note we're starting to explore the architecture of this system here: https://publiclab.org/notes/mridulnagpal/03-01-2017/map-of-projects

@mridulnagpal -- sorry I was in meetings much of last week. Looking now, I think if we can make that method available as a local method like these, that'd be great: https://github.com/publiclab/plots2/pull/1314/files#diff-21c89c193db238a944fc5e67377a6b01R182

We'll be using it to construct parameters in a query to save a location, make sense?

**Comment by mridulnagpal** [comment]
@jywarren I have to change the function everytime the map pans/moves thus I have to create it inside of addGrid() If I create it inside of it it won't be usable from outside thus I made it accessible through window. If the function is defined outside of addGid() I won't be able to access cells co-ordinates.What should I do?

**Comment by jywarren** [comment]
oh, i see -- can you not pass in another parameter? Or use a closure? I'll
take a look tonight or tomorrow and add some suggestions. Thanks!

**Comment by jywarren** [comment]
Oh, cool. Reviewing today, sorry i've been slow! Busy busy!

**Comment by jywarren** [comment]
OK almost there, and left some resources too. Thanks, Mridul!!!

**Comment by mridulnagpal** [comment]
@jywarren How do I add a power tag?

**Comment by jywarren** [comment]
Hi, @mridulnagpal - in tests or in the code, you can do something like:

```ruby
@node.add_tag('key:value', current_user)
```

But in the browser, you can enter `key:value` (or whatever your power tag is) in to the tags form. Make sense?

**Comment by mridulnagpal** [comment]
@jywarren Can you review the changes.

**Comment by mridulnagpal** [comment]
@jywarren Sorry I had my mid sems, could you please review the changes. I resolved the conflicts as well and will be maintaining my other repo side by side. Thanks

**Comment by jywarren** [comment]
Apologies - i had made some requests but in the new-ish "review" system i had never "submitted" them and so you never saw them! I'll be more careful next time.

**Comment by mridulnagpal** [comment]
@jywarren Done with the reviews please have a look.

**Comment by jywarren** [comment]
Oh, super -- i'll look over this tomorrow, and glad to have you back!

**Comment by jywarren** [comment]
Thanks, @mridulnagpal -- ok, i'm doing a careful read through of your plan and have some suggestions. Will post on that today! Merging this now, thanks!

**Pull request merged**